src/activity_actor.cpp

```diff
@@ -1482,10 +1482,12 @@ std::unique_ptr<activity_actor> consume_activity_actor::deserialize( JsonIn &jsi
     return actor.clone();
 }
 
-void try_sleep_activity_actor::start( player_activity &act, Character &/*who*/ )
+void try_sleep_activity_actor::start( player_activity &act, Character &who )
 {
     act.moves_total = to_moves<int>( duration );
     act.moves_left = act.moves_total;
+    who.set_movement_mode( move_mode_id( "prone" ) );
```

Comment:
When the player wake up, do they need to manually switch back to walking?

Reply:
Yes. Laying down to sleep is a conscious effort. Waking up is not. As such i think switching back to walking should be manual

Reply:
There are other actions that require conscious effort but is done automatically anyway, for example assignment of containers to inventory items. Not automatically switching back to walking might confuse players (especially those updating from an earlier version and is used to waking up in walking mode).

Reply:
I'm not hung up on it so I can add that, no problem.